src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -177,7 +178,22 @@
   "Retrieve the value of an 'If-None-Match' HTTP header from the supplied Ring
   request.  If the header is not found, returns nil."
   [request :- {schema/Keyword schema/Any}]
-  (rr/get-header request "If-None-Match"))
+  ;; SERVER-1153 - For a non-nil value, the characters '--gzip' will be
+  ;; stripped from the end of the value which is returned.  The '--gzip'
+  ;; characters are added by the Jetty web server to an HTTP Etag response
+  ;; header for cases in which the corresponding response payload has been
+  ;; gzipped.  Newer versions of Jetty, 9.3.x series, have logic in the
+  ;; GzipHandler to strip these characters back off of the If-None-Match header
+  ;; value before the Ring request would see it.  The version of Jetty being
+  ;; used at the time this code was written (9.2.10), however, did not have this
+  ;; logic to strip the "--gzip" characters from the incoming header.  This
+  ;; function compensates for that by stripping the characters here - before
+  ;; other Puppet Server code would use it. When/if Puppet Server is upgraded to
+  ;; a version of trapperkeeper-webserver-jetty9 which is based on Jetty 9.3.x
+  ;; or newer, it may be safe to take out the line that removes the '--gzip'
+  ;; characters.
+  (some-> (rr/get-header request "If-None-Match")
+          (str/replace #"--gzip$" "")))
```

Comment:
@rlinehan Sorry, I may not have provided enough detail for the question you were asking.

In Jetty 9.2.10 - and the latest Jetty, 9.3.7 - the specific characters "--gzip" will be added to the end of the original tag only when the corresponding response body has been gzipped.  I don't see anything in Jetty's current code that would lead to those extra characters appearing anywhere other than the end of the string.

See the following lines for where Jetty does this:

Jetty v9.2.10:

https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/gzip/GzipHttpOutput.java#L199
https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/AsyncGzipFilter.java#L135

Jetty 9.3.7: 

https://github.com/eclipse/jetty.project/blob/jetty-9.3.7.v20160115/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHttpOutputInterceptor.java#L218-L219
https://github.com/eclipse/jetty.project/blob/jetty-9.3.7.v20160115/jetty-http/src/main/java/org/eclipse/jetty/http/GzipHttpContent.java#L34

I don't think we're guaranteed that Jetty won't change its implementation later to do something different.  Although, it appears that just by using the GzipHandler, as we already do in latest trapperkeeper-webserver-jetty9 code, that we'll get some of the compensation for Etag / If-None-Match values "for free" as we upgrade to a newer Jetty version:

https://github.com/eclipse/jetty.project/blob/jetty-9.3.7.v20160115/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java#L440-L454

Also, the new integration test that I added for this PR should help catch a change in behavior if there is one in a newer Jetty.

I'd hope that the likelihood of Jetty changing this in a way that would break backward compatibility for upgrading users would be relatively low.  I've come across a fair amount of complaints from users to the Jetty project about the current behaviors and workarounds that they are doing to compensate for it and would hope that the Jetty maintainers would be mindful about this regarding any future changes they may decide to make in this area.